### PR TITLE
New RPC API endpoints

### LINKED ISSUE #7137: RPC API `Tokentx` returning different results
So we are currently making tests fetching data from testnets. 
We've tried with [Scroll Alpha](https://blockscout.scroll.io/) and  [Goerli](https://eth-goerli.blockscout.com/).
The issue comes when we try fetching data (usually from large smart contracts) using the query 
`?module=account&action=tokentx&address={addressHash}&start_block={startBlock}&end_block={endBlock}&offset=10000&page=1` . 
The problem we're encountering is that the response we get changes from one moment to the next, even within a span of 10 minutes.
Also, when we fetch from the same smart contract using `offset=1000` or another small number, the first 5 pages come empty and then the 6th has some transfers, which makes it difficult for us to know the total number of transfers.
We don't know if this is something we are doing wrong, so any help or advice would be greatly appreciated!

### Steps to reproduce

Using Goerli testnet:

1. Go to [this link](https://eth-goerli.blockscout.com/api?module=account&action=tokentx&address=0xc92470D7Ffa21473611ab6c6e2FcFB8637c8f330&start_block=8573412&end_block=8705096&offset=10000&page=1).
2. Open dev tools and switch to the network tab. Open the document response (you might need to reload the page) and look at the last index of the `result` array. At the moment of writing the last index is `4247`

With this smart contract you can also try changing the offset query to 1000 and you will notice that pages 1 to 5 are empty but the [page 6 currently has `210` results](https://eth-goerli.blockscout.com/api?module=account&action=tokentx&address=0xc92470D7Ffa21473611ab6c6e2FcFB8637c8f330&start_block=8573412&end_block=8705096&offset=1000&page=6)

Using Scroll Alpha testnet

1. Go to [this link](https://blockscout.scroll.io/api?module=account&action=tokentx&address=0xaC07EE38FA79d3Cc8d27E3e765f059b3b5D0dD8a&start_block=30224&offset=10000&page=0&end_block=526224) and follow the same steps from above. Right now the last index is `642` 

### Expected behaviour

Shouldn't offset give you the first X number of results from that query? If so what is happening with the first 6000 transfers from the link above?
Also, we expect the number of transfers to remain static. Not change unexpectedly.

### Actual behaviour

Responses vary from time to time and some pages come empty when they shouldn't

### PR DESCRIPTION
**Pull request by nikitosing**
#8847 
Closes #8868
Closes #7137 

## Changelog
- Account > ERC721 - Token Transfer Events
- Contract > Creator and creation tx hash
- Contract > Verify Source Code
- Contract > Verify Proxy Contract
- Do not return twin contracts in getsourcecode RPC API endpoint

https://github.com/blockscout/docs/pull/220
## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/explorer/priv/repo/migrations/20231227170848_add_proxy_verification_status.exs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+defmodule Explorer.Repo.Migrations.AddProxyVerificationStatus do
+  use Ecto.Migration
+
+  def change do
+    create table("proxy_verification_status", primary_key: false) do
+      add(:uid, :string, size: 64, primary_key: true)
+      add(:status, :int2, null: false)
+      add(:address_hash, :bytea, null: false)
```
Comment (vbaranov) [inline apps/explorer/priv/repo/migrations/20231227170848_add_proxy_verification_status.exs:8]: As we did in `tokens` table, I'd suggest to rename this field to `contract_address_hash`

File: apps/explorer/priv/repo/migrations/20231227170848_add_proxy_verification_status.exs
Comment (vbaranov) [inline apps/explorer/priv/repo/migrations/20231227170848_add_proxy_verification_status.exs:1]: We need to create a foreign key constraint on `smart_contracts` table with `ON DELETE CASCADE`.

File: apps/explorer/lib/explorer/chain/smart_contract.ex [hunk 2]
```diff
@@ -1204,6 +1201,18 @@ defmodule Explorer.Chain.SmartContract do
     end
   end
 
+  def get_fresh_time_distance do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/smart_contract.ex:1204]: Please add spec && doc for public function

File: apps/explorer/lib/explorer/chain/token_transfer.ex [hunk 3]
```diff
@@ -361,11 +360,19 @@ defmodule Explorer.Chain.TokenTransfer do
...
     from(token_transfer in __MODULE__,
-      inner_join: block in Block,
-      on: token_transfer.block_hash == block.hash,
+      inner_join: block in assoc(token_transfer, :block),
+      as: :block,
       where: block.consensus == true
     )
   end
+
+  def erc_721_token_transfers_query do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/token_transfer.ex:372]: Please add spec && doc for public function

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex [hunk 4]
```diff
@@ -185,6 +188,31 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
...
+         true <- !is_nil(address_hash) or !is_nil(contract_address_hash),
+         {:ok, token_transfers, max_block_number} <-
+           list_nft_token_transfers(address_hash, contract_address_hash, options) do
+      render(conn, :tokennfttx, %{token_transfers: token_transfers, max_block_number: max_block_number})
+    else
+      false ->
+        render(conn, :error, error: "Query parameter address or contractaddress is required")
+
+      {:address, :error} ->
+        render(conn, :error, error: "Invalid address format")
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex:206]: Let's define module attribute. It is repeated multiple times in this module.

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex [hunk 5]
```diff
@@ -185,6 +188,31 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
...
+
+      {:contract_address, :error} ->
+        render(conn, :error, error: "Invalid contract address format")
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex:209]: Let's define module attribute. It is repeated multiple times in this module.

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex [hunk 6]
```diff
@@ -185,6 +188,31 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
...
+
+      {_, :not_found} ->
+        render(conn, :error, error: "No token transfers found", data: [])
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex:212]: Let's define module attribute. It is repeated multiple times in this module.

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/helper.ex [hunk 7]
```diff
@@ -39,4 +39,19 @@ defmodule BlockScoutWeb.API.RPC.Helper do
   defp validate_max_page_size(page_size) do
     if page_size <= Etherscan.page_size_max(), do: :ok, else: :error
   end
+
+  @spec parse_and_validate_addresses(binary(), integer()) :: list()
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/helper.ex:43]: doc is missing

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_controller.ex [hunk 8]
```diff
@@ -497,7 +497,12 @@ defmodule BlockScoutWeb.API.V2.AddressController do
     end
   end
 
-  defp validate_address(address_hash_string, params, options \\ @api_true) do
+  @spec validate_address(binary(), any(), any()) ::
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_controller.ex:500]: doc is missing

File: apps/explorer/lib/explorer/chain/smart_contract.ex [hunk 9]
```diff
@@ -1179,22 +1185,13 @@ defmodule Explorer.Chain.SmartContract do
   defp db_implementation_data_converter(string) when is_binary(string), do: string
   defp db_implementation_data_converter(other), do: to_string(other)
 
-  defp check_implementation_refetch_necessity(nil), do: true
+  def check_implementation_refetch_necessity(nil), do: true
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/smart_contract.ex:1188]: Spec && doc

File: apps/explorer/lib/explorer/chain/smart_contract.ex [hunk 10]
```diff
@@ -523,7 +523,7 @@ defmodule Explorer.Chain.SmartContract do
         smart_contract
       end
 
-    get_implementation_address_hash({:updated, updated_smart_contract})
+    get_implementation_address_hash({:updated, updated_smart_contract}, options)
   end
 
   def get_implementation_address_hash(
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/smart_contract.ex:529]: Let's add spec && doc

File: apps/explorer/lib/explorer/chain/token_transfer.ex [hunk 11]
```diff
@@ -361,11 +360,19 @@ defmodule Explorer.Chain.TokenTransfer do
 
   def filter_by_type(query, _), do: query
 
+  @spec only_consensus_transfers_query() :: Ecto.Query.t()
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/token_transfer.ex:363]: Please add doc

File: apps/explorer/lib/explorer/etherscan/contracts.ex [hunk 12]
```diff
@@ -17,7 +17,7 @@ defmodule Explorer.Etherscan.Contracts do
   alias Explorer.Chain.SmartContract.Proxy.EIP1167
 
   @spec address_hash_to_address_with_source_code(Hash.Address.t()) :: Address.t() | nil
-  def address_hash_to_address_with_source_code(address_hash) do
+  def address_hash_to_address_with_source_code(address_hash, twin_needed? \\ true) do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/etherscan/contracts.ex:20]: Please add doc

File: apps/explorer/lib/explorer/etherscan.ex [hunk 13]
```diff
@@ -257,6 +259,44 @@ defmodule Explorer.Etherscan do
...
+        contract_address_hash,
+        options \\ @default_options
+      ) do
+    options
+    |> base_nft_token_transfers_query(contract_address_hash)
+    |> where([tt], tt.from_address_hash == ^address_hash or tt.to_address_hash == ^address_hash)
+    |> Repo.replica().all()
+  end
+
+  @spec list_nft_token_transfers_by_token(Hash.Address.t(), map()) :: [TokenTransfer.t()]
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/etherscan.ex:274]: Please add doc

File: apps/explorer/lib/explorer/etherscan.ex [hunk 14]
```diff
@@ -257,6 +259,44 @@ defmodule Explorer.Etherscan do
     end
   end
 
+  @spec list_nft_token_transfers(Hash.Address.t(), Hash.Address.t() | nil, map()) :: [TokenTransfer.t()]
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/etherscan.ex:262]: Please add doc

File: apps/explorer/lib/explorer/chain.ex [hunk 15]
```diff
@@ -4396,10 +4396,10 @@ defmodule Explorer.Chain do
       :not_found
 
   """
-  @spec check_address_exists(Hash.Address.t()) :: :ok | :not_found
-  def check_address_exists(address_hash) do
+  @spec check_address_exists(Hash.Address.t(), [api?]) :: :ok | :not_found
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain.ex:4399]: Does it make sense to move the function to Explorer.Chain.Address module?

File: apps/explorer/lib/explorer/chain.ex [hunk 16]
```diff
@@ -4421,15 +4421,15 @@ defmodule Explorer.Chain do
       false
 
   """
-  @spec address_exists?(Hash.Address.t()) :: boolean()
-  def address_exists?(address_hash) do
+  @spec address_exists?(Hash.Address.t(), [api?]) :: boolean()
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain.ex:4424]: Does it make sense to move the function to Explorer.Chain.Address module?

File: apps/explorer/lib/explorer/chain/smart_contract/proxy/verification_status.ex [hunk 17]
```diff
@@ -0,0 +1,94 @@
+defmodule Explorer.Chain.SmartContract.Proxy.VerificationStatus do
```
Comment (sl1depengwyn) [inline apps/explorer/lib/explorer/chain/smart_contract/proxy/verification_status.ex:1]: Please, add specs and docs for public functions

File: apps/explorer/lib/explorer/chain/smart_contract.ex (hunk 10 above)
Comment (nikitosing) [reply apps/explorer/lib/explorer/chain/smart_contract.ex:529]: doc and spec for `get_implementation_address_hash/2` already exist

**Review by vbaranov — COMMENTED** [review]
Let's handle the case when required parameters were not passed like here when calling `/api?module=contract&action=verifysourcecode`:
```
2024-01-10T12:51:22.949 application=api request_id=F6jzlaFuqpFiSm0AAAZC [error] Error while calling RPC action"** (Phoenix.ActionClauseError) no function clause matching in BlockScoutWeb.API.RPC.ContractController.verifysourcecode/2\n\nThe following arguments were given to BlockScoutWeb.API.RPC.ContractController.verifysourcecode/2:\n\n    # 1\n    %Plug.Conn{adapter: {Plug.Cowboy.Conn, :...}, assigns: %{}, body_params: %{}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: \"localhost\", method: \"POST\", owner: #PID<0.1798.0>, params: %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"}, path_info: [], path_params: %{}, port: 4000, private: %{BlockScoutWeb.ApiRouter => {[\"api\"], %{Absinthe.Plug => [\"graphql\"], BlockScoutWeb.API.RPC.RPCTranslator => [], BlockScoutWeb.APIKeyV2Router => [\"v2\", \"key\"], BlockScoutWeb.SmartContractsApiV2Router => [\"v2\", \"smart-contracts\"], RPCTranslatorForwarder => []}}, BlockScoutWeb.Router => {[], %{Absinthe.Plug.GraphiQL => [\"graphiql\"], BlockScoutWeb.ApiRouter => [\"api\"], BlockScoutWeb.WebRouter => []}}, :before_send => [#Function<0.117693457/1 in BlockScoutWeb.Plug.Logger.call/2>, #Function<1.71982575/1 in Phoenix.LiveReloader.before_send_inject_reloader/3>, #Function<2.86362936/1 in BlockScoutWeb.Endpoint.call/2>], :phoenix_action => :verifysourcecode, :phoenix_controller => BlockScoutWeb.API.RPC.ContractController, :phoenix_endpoint => BlockScoutWeb.Endpoint, :phoenix_format => \"json\", :phoenix_layout => {BlockScoutWeb.LayoutView, :app}, :phoenix_router => BlockScoutWeb.ApiRouter, :phoenix_view => BlockScoutWeb.API.RPC.ContractView, :plug_session_fetch => #Function<1.84243074/1 in Plug.Session.fetch_session/1>}, query_params: %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"}, query_string: \"module=contract&action=verifysourcecode\", remote_ip: {127, 0, 0, 1}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{\"accept\", \"*/*\"}, {\"accept-encoding\", \"gzip, deflate, br\"}, {\"cache-control\", \"no-cache\"}, {\"connection\", \"keep-alive\"}, {\"content-length\", \"0\"}, {\"host\", \"localhost:4000\"}, {\"postman-token\", \"7e6c1b7f-f860-4f14-82b6-b49bdaf7e880\"}, {\"user-agent\", \"PostmanRuntime/7.36.0\"}], request_path: \"/api\", resp_body: nil, resp_cookies: %{}, resp_headers: [{\"cache-control\", \"max-age=0, private, must-revalidate\"}, {\"x-request-id\", \"F6jzlaFuqpFiSm0AAAZC\"}, {\"access-control-allow-credentials\", \"true\"}, {\"access-control-allow-origin\", \"*\"}, {\"access-control-expose-headers\", \"\"}], scheme: :http, script_name: [\"api\"], secret_key_base: :..., state: :unset, status: nil}\n\n    # 2\n    %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"}\n\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:146: BlockScoutWeb.API.RPC.ContractController.verifysourcecode(%Plug.Conn{adapter: {Plug.Cowboy.Conn, :...}, assigns: %{}, body_params: %{}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: \"localhost\", method: \"POST\", owner: #PID<0.1798.0>, params: %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"}, path_info: [], path_params: %{}, port: 4000, private: %{BlockScoutWeb.ApiRouter => {[\"api\"], %{Absinthe.Plug => [\"graphql\"], BlockScoutWeb.API.RPC.RPCTranslator => [], BlockScoutWeb.APIKeyV2Router => [\"v2\", \"key\"], BlockScoutWeb.SmartContractsApiV2Router => [\"v2\", \"smart-contracts\"], RPCTranslatorForwarder => []}}, BlockScoutWeb.Router => {[], %{Absinthe.Plug.GraphiQL => [\"graphiql\"], BlockScoutWeb.ApiRouter => [\"api\"], BlockScoutWeb.WebRouter => []}}, :before_send => [#Function<0.117693457/1 in BlockScoutWeb.Plug.Logger.call/2>, #Function<1.71982575/1 in Phoenix.LiveReloader.before_send_inject_reloader/3>, #Function<2.86362936/1 in BlockScoutWeb.Endpoint.call/2>], :phoenix_action => :verifysourcecode, :phoenix_controller => BlockScoutWeb.API.RPC.ContractController, :phoenix_endpoint => BlockScoutWeb.Endpoint, :phoenix_format => \"json\", :phoenix_layout => {BlockScoutWeb.LayoutView, :app}, :phoenix_router => BlockScoutWeb.ApiRouter, :phoenix_view => BlockScoutWeb.API.RPC.ContractView, :plug_session_fetch => #Function<1.84243074/1 in Plug.Session.fetch_session/1>}, query_params: %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"}, query_string: \"module=contract&action=verifysourcecode\", remote_ip: {127, 0, 0, 1}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{\"accept\", \"*/*\"}, {\"accept-encoding\", \"gzip, deflate, br\"}, {\"cache-control\", \"no-cache\"}, {\"connection\", \"keep-alive\"}, {\"content-length\", \"0\"}, {\"host\", \"localhost:4000\"}, {\"postman-token\", \"7e6c1b7f-f860-4f14-82b6-b49bdaf7e880\"}, {\"user-agent\", \"PostmanRuntime/7.36.0\"}], request_path: \"/api\", resp_body: nil, resp_cookies: %{}, resp_headers: [{\"cache-control\", \"max-age=0, private, must-revalidate\"}, {\"x-request-id\", \"F6jzlaFuqpFiSm0AAAZC\"}, {\"access-control-allow-credentials\", \"true\"}, {\"access-control-allow-origin\", \"*\"}, {\"access-control-expose-headers\", \"\"}], scheme: :http, script_name: [\"api\"], secret_key_base: :..., state: :unset, status: nil}, %{\"action\" => \"verifysourcecode\", \"module\" => \"contract\"})\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:1: BlockScoutWeb.API.RPC.ContractController.action/2\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:1: BlockScoutWeb.API.RPC.ContractController.phoenix_controller_pipeline/2\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/rpc_translator.ex:119: BlockScoutWeb.API.RPC.RPCTranslator.call_controller/3\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/rpc_translator.ex:37: BlockScoutWeb.API.RPC.RPCTranslator.call/2\n    (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2\n    (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2\n    (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2\n    (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2\n    (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.plug_builder_call/2\n    (block_scout_web 6.0.0) /Users/v/blockscout/deps/plug/lib/plug/debugger.ex:136: BlockScoutWeb.Endpoint.\"call (overridable 3)\"/2\n    (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.\"call (overridable 4)\"/2\n    (block_scout_web 6.0.0) /Users/v/blockscout/deps/spandex_phoenix/lib/spandex_phoenix.ex:151: BlockScoutWeb.Endpoint.call/2\n    (phoenix 1.5.14) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3\n    (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3\n
```

**Comment by nikitosing** [comment]
@vbaranov Thanks! Done in https://github.com/blockscout/blockscout/pull/9068/commits/fcf6705db12c367d8f9e654c43b8c333128a1dc9

**Review by vbaranov — COMMENTED** [review]
Error in checkproxyverification endpoint
http://localhost:4000/api/v1?module=contract&action=checkproxyverification&guid=5267789f4b7b4efd7bcb235d045080b9798ad87c659e7792

```
2024-01-10T13:59:41.047 application=api request_id=F6j3T8tJAkuQ_aQAAPPD [error] Error while calling RPC action"** (Plug.Conn.WrapperError) ** (KeyError) key :status not found in: nil. If you are using the dot syntax, such as map.field, make sure the left-hand side of the dot is a map\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:264: BlockScoutWeb.API.RPC.ContractController.checkproxyverification/2\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:1: BlockScoutWeb.API.RPC.ContractController.action/2\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/contract_controller.ex:1: BlockScoutWeb.API.RPC.ContractController.phoenix_controller_pipeline/2\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/rpc_translator.ex:119: BlockScoutWeb.API.RPC.RPCTranslator.call_controller/3\n    (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/rpc/rpc_translator.ex:37: BlockScoutWeb.API.RPC.RPCTranslator.call/2\n    (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2\n    (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2\n    (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2\n    (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2\n    (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.plug_builder_call/2\n    (block_scout_web 6.0.0) /Users/v/blockscout/deps/plug/lib/plug/debugger.ex:136: BlockScoutWeb.Endpoint.\"call (overridable 3)\"/2\n    (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.\"call (overridable 4)\"/2\n    (block_scout_web 6.0.0) /Users/v/blockscout/deps/spandex_phoenix/lib/spandex_phoenix.ex:151: BlockScoutWeb.Endpoint.call/2\n    (phoenix 1.5.14) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3\n    (cowboy 2.10.0) /Users/v/blockscout/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3\n    (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3\n"
```

**Comment by nikitosing** [comment]
@vbaranov nice catch! fixed in https://github.com/blockscout/blockscout/pull/9068/commits/ff1016631dffcc72058f8e7b8db305ef3e86abee

**Review by vbaranov — APPROVED** [review]

**Review by sl1depengwyn — APPROVED** [review]

**Pull request merged**